sycl/test-e2e/ESIMD/rdtsc.cpp

```diff
@@ -60,13 +58,9 @@ int test_rdtsc_sr0() {
 
   int Result = 0;
 
-  // Check if returned values are not the same
-  std::sort(VectorOutputRDTSC.begin(), VectorOutputRDTSC.end());
-  std::sort(VectorOutputSR0.begin(), VectorOutputSR0.end());
-  Result |= std::equal(VectorOutputRDTSC.begin() + 1, VectorOutputRDTSC.end(),
-                       VectorOutputRDTSC.begin());
-  Result |= std::equal(VectorOutputSR0.begin() + 1, VectorOutputSR0.end(),
-                       VectorOutputSR0.begin());
+  // Check if returned values are positive
+  Result |= std::any_of(VectorOutputRDTSC.begin(), VectorOutputRDTSC.end(),
+                        [](uint64_t v) { return v <= 0; });
```

Comment:
if `v` is unsigned, then how it can be less than zero?
The `VectorOutputRDTSC` needs to be signed.

Reply:
If suppose 2Ghz, then 2 secs = 32-bit. the timestamp returns 64-bit, which is huge (2 sec * 4Bil, so, there was no risk of overflow). The updated solution is good, except it needs 1 minor fix to avoid DCE-ing the load between rdtsc calls.